messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/AbstractMessagingDatabusConsumer.java

```diff
@@ -299,6 +295,15 @@ protected String getReaderNameSuffix() {
 
   protected abstract void createPartitionReaders() throws IOException;
 
+  public synchronized Long getPendingData() throws Exception {
+    Long pendingSize = 0l;
+    //the backlog will the summation of backlog of all the individual readers
+    for (Iterator<PartitionReader> it = readers.values().iterator();it.hasNext();) {
```

Comment:
you could simplify by using foreach loop here.

for (PartitionReader reader : readers.values()) {
  pendingSize += reader.getReaderBacklog();
}